### PR TITLE
Add environment variable expansion in external label values

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -1164,14 +1166,35 @@ func TestBadStaticConfigsYML(t *testing.T) {
...
 }
 
+func TestExpandExternalLabels(t *testing.T) {
+	// Cleanup ant TEST env variable that could exist on the system.
+	os.Setenv("TEST", "")
+
+	c, err := LoadFile("testdata/external_labels.good.yml", false, log.NewNopLogger())
+	require.NoError(t, err)
+	require.Equal(t, labels.Label{Name: "bar", Value: "foo"}, c.GlobalConfig.ExternalLabels[0])
+	require.Equal(t, labels.Label{Name: "foo", Value: "${TEST}"}, c.GlobalConfig.ExternalLabels[1])
```
Comment (beorn7) [inline config/config_test.go:1182]: Perhaps also add one with non-variable text, e.g. "foo${TEST}bar"

**Review by beorn7 — APPROVED** [review]
Just an idea for an additional test case.

**Comment by roidelapluie** [comment]
Note: This is pending a vote: https://groups.google.com/u/3/g/prometheus-developers/c/4JmD_glv0zg

**Review by bwplotka — APPROVED** [review]
LGTM 👍🏽 

Thanks

**Pull request merged**